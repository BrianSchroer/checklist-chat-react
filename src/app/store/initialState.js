import * as modalDialogType from '../modalDialogType';

export default {
    rooms: [],
    currentRoomId: null,
    ajaxCallsInProgressCount: 0,
    modalDialog: modalDialogType.NONE
};
