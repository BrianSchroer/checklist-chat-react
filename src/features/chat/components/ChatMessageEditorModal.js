import React, {PropTypes} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import * as chatMessageType from '../chatMessageType';
import {saveChatMessage} from '../chatDucks';
import {validate} from '../chatMessageValidator';
import Modal from '../../../components/Modal';
import TextInput from '../../../components/TextInput';

export class ChatMessageEditorModal extends React.Component {
    constructor(props, context) {
        super(props, context);

        this.state = {
            chatMessage: Object.assign({}, props.chatMessage),
            isSaving: false,
            isDirty: false,
            errors: {}
        };

        this.onChange = this.onChange.bind(this);
        this.onSave = this.onSave.bind(this);
    }

    onChange(event) {
        const chatMessage = this.state.chatMessage;
        chatMessage[event.target.name] = event.target.value;
        return this.setState({chatMessage: chatMessage, isDirty: true});
    }

    onSave(event) {
        event.preventDefault();

        const {userId, roomId} = this.props;
        const {chatMessage} = this.state;

        const validationResponse = validate(chatMessage);

        if (!validationResponse.isValid) {
            this.setState({errors: validationResponse.errors});
            return;
        }

        const {actions, onCloseRequest} = this.props;
        actions.saveChatMessage(chatMessage, roomId, userId);
        onCloseRequest(event);
    }

    render() {
        const {onCloseRequest} = this.props;
        const {chatMessage, errors} = this.state;

        const buttons = (
            <div>
                <input type="button" value="Cancel" className="btn btn-default"
                    onClick={onCloseRequest}/>
                <input type="submit" value="Save" className="btn btn-primary"
                    onClick={this.onSave}/>
            </div>
        );

        return (
            <Modal title="Say Something..." onCloseRequest={onCloseRequest} buttons={buttons}>

                <TextInput name="text" label="Message"
                    rows={3} value={chatMessage.text}
                    onChange={this.onChange} error={errors.text} />

            </Modal>
        );
    }
}

ChatMessageEditorModal.propTypes = {
    roomId: PropTypes.string,
    userId: PropTypes.string.isRequired,
    chatMessage: PropTypes.object.isRequired,
    onCloseRequest: PropTypes.func.isRequired,
    actions: PropTypes.object.isRequired
};

function mapStateToProps(state, ownProps) {
    const userId = state.userId;
    const roomId = state.roomId;

    const chatMessage = {
        chatMessageType: chatMessageType.CHAT,
        text: ''
    };

    const onCloseRequest = ownProps.onCloseRequest;

    return {userId, roomId, chatMessage, onCloseRequest};
}

const mapDispatchToProps = (dispatch) => ({
    actions: bindActionCreators({saveChatMessage}, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(ChatMessageEditorModal);
