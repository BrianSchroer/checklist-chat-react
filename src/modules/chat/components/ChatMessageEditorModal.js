import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {chatMessageType} from '../../chat';
import {saveChatMessage} from '../chatDucks';
import {validate} from '../chatMessageValidator';
import {Modal, TextInput} from '../../../modules/core';

export class ChatMessageEditorModal extends React.Component {

    static get propTypes() {
        return {
            roomId: PropTypes.string.isRequired,
            onCloseRequest: PropTypes.func.isRequired,
            actions: PropTypes.object.isRequired
        };
    }

    constructor(props, context) {
        super(props, context);

        this.state = {
            chatMessage: {
                chatMessageType: chatMessageType.CHAT,
                text: ''
            },
            isSaving: false,
            isDirty: false,
            errors: {}
        };

        this.onChange = this.onChange.bind(this);
        this.onSave = this.onSave.bind(this);
    }

    onChange(event) {
        const chatMessage = this.state.chatMessage;
        chatMessage[event.target.name] = event.target.value;
        return this.setState({chatMessage: chatMessage, isDirty: true});
    }

    onSave(event) {
        event.preventDefault();

        const {roomId} = this.props;
        const {chatMessage} = this.state;

        const validationResponse = validate(chatMessage);

        if (!validationResponse.isValid) {
            this.setState({errors: validationResponse.errors});
            return;
        }

        const {actions, onCloseRequest} = this.props;
        actions.saveChatMessage(chatMessage, roomId);
        onCloseRequest(event);
    }

    render() {
        const {onCloseRequest} = this.props;
        const {chatMessage, errors} = this.state;

        const buttons = (
            <div>
                <input type="button" value="Cancel" className="btn btn-default"
                    onClick={onCloseRequest}/>
                <input type="submit" value="Submit" className="btn btn-primary" />
            </div>
        );

        return (
            <Modal
                title="Say Something..."
                onSubmit={this.onSave}
                onCloseRequest={onCloseRequest}
                buttons={buttons}>

                <TextInput name="text" label="Message"
                    rows={3} value={chatMessage.text}
                    onChange={this.onChange} error={errors.text} />

            </Modal>
        );
    }
}

export function mapStateToProps(state, ownProps) {
    const {roomId} = state;

    const onCloseRequest = ownProps.onCloseRequest;

    return {roomId, onCloseRequest};
}

const mapDispatchToProps = (dispatch) => ({
    actions: bindActionCreators({saveChatMessage}, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(ChatMessageEditorModal);
