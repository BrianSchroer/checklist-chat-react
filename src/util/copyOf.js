export default original => Object.assign({}, original);
