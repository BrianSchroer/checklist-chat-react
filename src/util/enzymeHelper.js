const enzymeHelper = class {

    static nestedFind(wrapper, selector) {
        const selectors = selector.split('>').map(node => node.trim());
        let parent = wrapper;
        let found = [];

        for (let i = 0; i < selectors.length; i++) {
            found = parent.find(selectors[i]);
            if (!found.length) {
                break;
            }
        }

        return found;
    }

    static findSingle(wrapper, selector) {
        const findArray = enzymeHelper.nestedFind(wrapper, selector);

        if (findArray.length != 1) {
            throw(`expected 1 find for selector "${selector}", but found ${findArray.length}.`);
        }

        return findArray.first();
    }
};

export default enzymeHelper;
