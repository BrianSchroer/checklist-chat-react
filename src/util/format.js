import moment from 'moment';

export default class {
    static time(input = '') {

        const isoDateFormatRegex = /\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}([+-]\d{2}:\d{2})?/g;
        let formatted = input;

        if (input && input.match(isoDateFormatRegex)) {
            formatted = this.dateTimeString(input, 'h:mm a');
        }

        return formatted;
    }

    static dateTimeString(input, formatString) {
        let formatted = input;

        if (input) {
            const testMoment = moment(input);

            if (testMoment.isValid()) {
                const testFormat = testMoment.format(formatString);
                if (testFormat != 'Invalid date') {
                    formatted = testFormat;
                }
            }
        }

        return formatted;
    }
}
