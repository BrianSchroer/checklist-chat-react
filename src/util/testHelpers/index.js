import {shallow, mount} from 'enzyme';
import enzymeHelper from './enzymeHelper';

export {
    shallow as shallow,
    mount as mount,
    enzymeHelper as enzymeHelper
};
