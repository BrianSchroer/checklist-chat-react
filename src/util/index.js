import format from './format';
import uiHelper from './uiHelper';
import validationHelper from './validationHelper';

export {
    format as format,
    uiHelper as uiHelper,
    validationHelper as validationHelper
};
