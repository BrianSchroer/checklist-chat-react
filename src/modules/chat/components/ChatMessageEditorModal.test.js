import React from 'react';
import initialState from '../../../app/store/initialState';
import {snapshotHelper} from '../../../util/testHelpers';
import {ChatMessageEditorModal, mapStateToProps} from './ChatMessageEditorModal';

const defaultProps = {
    roomId: '123',
    onCloseRequest: () => {},
    actions: {}
};

const defaultOwnProps = {
    onCloseRequest: () => {}
};

function overrideProps(propOverrides) {
    return Object.assign({}, defaultProps, propOverrides);
}

function assertSnapshotMatch(propOverrides = {}) {
    snapshotHelper.assertMatch(<ChatMessageEditorModal {...overrideProps(propOverrides)} />);
}

function callMapStateToProps(stateOverrides, ownPropsOverrides) {
    const state = Object.assign({}, initialState, stateOverrides || {});
    const ownProps = Object.assign({}, defaultOwnProps, ownPropsOverrides || {});

    return mapStateToProps(state, ownProps);
}

describe('ChatMessageEditorModal', () => {
    it('should render correctly', () => {
        assertSnapshotMatch();
    });

    describe('mapStateToProps', () => {
        it('should return expected props', () => {
            const stateOverrides = { roomId: 666 };
            const props = callMapStateToProps(stateOverrides);

            expect(props.roomId).toBe(stateOverrides.roomId);
            expect(props.onCloseRequest).toBe(defaultOwnProps.onCloseRequest);
        });
    });
});
