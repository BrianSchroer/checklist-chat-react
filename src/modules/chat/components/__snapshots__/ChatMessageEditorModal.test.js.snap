// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatMessageEditorModal should render correctly 1`] = `
<div
  className="checklist-chat-modal-backdrop"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog checklist-chat-modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="modal-header bg-primary text-white"
        >
          <button
            aria-label="Close"
            className="close"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
            >
              Ã—
            </span>
          </button>
          <h4
            className="modal-title display-linebreaks"
          >
            Say Something...
          </h4>
        </div>
        <div
          className="modal-body"
        >
          <div
            className="form-group"
          >
            <label
              htmlFor="text"
            >
              Message
            </label>
            <div
              className="field"
            >
              <textarea
                className="form-control"
                name="text"
                onChange={[Function]}
                placeholder={undefined}
                rows={3}
                value=""
              />
            </div>
          </div>
        </div>
        <div
          className="modal-footer"
        >
          <div>
            <input
              className="btn btn-default"
              onClick={[Function]}
              type="button"
              value="Cancel"
            />
            <input
              className="btn btn-primary"
              type="submit"
              value="Submit"
            />
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
